File: Cracking Coding Interview/Through Java/LearningToSolving/src/Course/Module_4/Assignment/Task_2.java
package Course.Module_4.Assignment;

import java.util.Scanner;

public class Task_2 {
    // Solution of the task 2
    int solution(int a, int b) {
        if (b < 1) {
            return 0;
        }
        return a+solution(a, b-1);
    }

    // Main method
    public static void main(String[] args) {
        // User input
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the first number: ");
        int a = sc.nextInt();
        System.out.print("Enter the second number: ");
        int b = sc.nextInt();

        Task_2 obj1 = new Task_2();
        System.out.println("The result of multiplication: " + obj1.solution(a, b));
    }
}


// Time Complexity : O(n)
// Space Complexity : O(n)

File: Cracking Coding Interview/Through C++/LearningToSolving/Course/Module-4/Assignment/Task_2.cpp
#include <iostream>

using namespace std;

class Task_2 {
    public:
        // Solution method
        int solution(int cns, int cng) {
            if (cng < 1) {
                return 0;
            }

            return cns + solution(cns, cng - 1);
        }
};

// Main method
int main() {
    int a, b, cns, cng;

    // Take the user input
    cout << "Enter the first number: ";
    cin >> a;
    cout << "Enter the second number: ";
    cin >> b;

    if (a > b) {
        cns = a;
        cng = b;
    } else {
        cns = b;
        cng = a;
    }

    // Object instantiation
    Task_2 obj;
    int fr = obj.solution(cns, cng);

    // Print the final result
    cout << "The result of multiplication of " << a << " and " << b << " : " << fr << "\n";

    return 0;
}

/*
    Time Complexity: O(n)
    Space Complexity: O(n)
*/